src/List.js

```diff
@@ -0,0 +1,159 @@
+            [field]: value
+        })
+    }
+
+    function handleNewTask(title, priority){
+        const taskId = generateUniqueID()
+        setDoc(doc(tasksCollectionRef, taskId), {
+            id: taskId,
+            title: title,
+            completed: false,
+            created: serverTimestamp(),
+            priority: priority
+        })
+    }
+
+    function handleDeleteFinished(){
+        let completed = tasks.filter(task => task.completed)
+        completed.forEach(task => deleteDoc(doc(tasksCollectionRef, task.id)))
+    }
+
+    function renameList(name) {
+        updateDoc(listDocRef, {
+            name: name
+        })
+    }
+
+    function deleteList() {
+        tasks.forEach(task => deleteDoc(doc(tasksCollectionRef, task.id)))
+        deleteDoc(doc(collection(props.db, 'Lists'), list.id))
+        props.setTab(list.id)
+    }
+
+    const TasksToDisplay = () => {
+        const allTasks = [...tasks]
+        const completed = allTasks.filter(task => task.completed)
+        const uncompleted = allTasks.filter(task => !task.completed)
+
+        const tasksToDisplay = showCompleted ? uncompleted.concat(completed) : uncompleted
+
+        return (
+            <>
+            {tasksToDisplay.length===0 && (
+                <Typography>No tasks for display :)</Typography>
+            )}
+            {tasksToDisplay.length!==0 && (
+                tasksToDisplay.map(task => (
+                <Task
+                    key ={task.id}
+                    {...task}
+                    deleteSingle={deleteSingle}
+                    handleEditTask={handleEditTask}
+                />
+            )))}
+            </>
+        )
+    }
+
+    const NavbarTasks = () => {
+        return (
+            <Navbar
+                handleNewTask={handleNewTask}
+                showCompleted={list.showCompleted}
+                onToggleComplete={onToggleComplete}
+                handleDeleteFinished={handleDeleteFinished}
+                onChangeSortOption={onChangeSortOption}
+                onChangeSortDirection={onChangeSortDirection}
+                sortBy={list.sortBy}
+                sortDir={list.sortDir}
+                renameList={renameList}
+                deleteList={deleteList}
+                list={list}
+            />
+        )
+    }
+    
+
+    if (loading) {
+        return (
+            <>
+                <NavbarTasks />
+                <p>Loading</p>
+            </>
+        )
+    }else if (error) {
+        return (
+            <p>Error: {JSON.stringify(error)}</p>
+        )
+    }else if (tasks){
+        return (
+            <>
+                <NavbarTasks />
+                <TasksToDisplay />
+            </>
+        )
+    }
+
+}
+
+export default function List(props) {
+    const app = props.app;
+    const db = getFirestore(app);
+    const listDocRef = doc(db, `Lists/${props.listId}`);
+    const [list, loading, error] = useDocument(listDocRef);
```

Comment:
If you made the parent pass in the list document (which it has already loaded) rather than just the listID, you wouldn't have to call useDocument to re-retrieve the document